### PR TITLE
Hide credential information in replication document for reader

### PR DESCRIPTION
**Pull request by jiangphcn**
## Overview

Replication document is available for user to check replication status. It includes some credential information. Before this PR, the `oauth` field is removed when replication status is displayed to user.  However, for basic authentication, the credential information is also saved in `headers` field and exposed to users. The aim of this PR is to hide credential information even if customer uses basic authentication case.

```

  "source": {
    "headers": {
      "Authorization": *****
    },
    "url": "https://testy011-user1.cloudant.com/testy_db_ctkzwqshsk"
  },
  "target": {
    "headers": {
      "Authorization": *****
    },
    "url": "https://testy011-user1.cloudant.com/test-db-auth"
  },
```

## Testing recommendations

make check skip_deps+=couch_epi apps=couch_replicator
```
module 'couch_replicator_docs'
couch_replicator_docs: check_options_pass_values_test...ok
couch_replicator_docs: check_options_fail_values_test...ok
couch_replicator_docs: check_convert_options_pass_test...ok
couch_replicator_docs: check_convert_options_fail_test...ok
couch_replicator_docs: check_strip_credentials_test...ok
[done in 0.015 s]
```

## Related Issues or Pull Requests

N/A

## Checklist

- [X] Code is written and works correctly;
- [X] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

File: src/couch_replicator/src/couch_replicator_docs.erl [hunk 1]
```diff
@@ -695,7 +695,8 @@ strip_credentials(Url) when is_binary(Url) ->
         "http\\1://\\2",
         [{return, binary}]);
 strip_credentials({Props}) ->
-    {lists:keydelete(<<"oauth">>, 1, Props)}.
+    Props0 = lists:keydelete(<<"oauth">>, 1, Props),
```
Comment (nickva) [inline src/couch_replicator/src/couch_replicator_docs.erl]: Minor nit on variable names --  `Props1` might work a bit better here.

Usually when adding new code towards the middle or the bottom of the function, start adding 1,2,3 like so `Props, Props1=newoperation(Props)..,  ...use Props1 from now on...`. If there is a large function that uses `Props` in the body and we need to do modify it at the beginning of the function, then it's better to do ```function(Prop0 =...) -> Props = newoperation(Props0), ....```

Comment (nickva) [inline src/couch_replicator/src/couch_replicator_docs.erl]: There are a few unit tests below, let's add a few tests for the new code as well.

Comment (nickva) [inline src/couch_replicator/src/couch_replicator_docs.erl]: Let's remove the bugzid bit from the commit as well.

**Comment by nickva** [comment]
Looks good. First I thought it was a bit heavy handed to remove all headers but it's probably safer. See some requested changes then +1

File: src/couch_replicator/src/couch_replicator_docs.erl (hunk 1 above)
Comment (jiangphcn) [reply src/couch_replicator/src/couch_replicator_docs.erl]: Thanks for your tips. Changed from `Props0` to `Props1`.

Comment (jiangphcn) [reply src/couch_replicator/src/couch_replicator_docs.erl]: Yes, added `check_strip_credentials_test`

Comment (jiangphcn) [reply src/couch_replicator/src/couch_replicator_docs.erl]: Yes, removed.

**Pull request merged**